tools/ctl/ctl.c

```diff
@@ -518,6 +531,11 @@ static int ctl_set_get(struct ctl_data *ctl_data)
 			fprintf(stderr, "Error: failed TLV read.\n");
 			return ret;
 		}
+		/* Replace the numid value (scontrol->comp_id) with SOF_CTRL_CMD_BINARY
+		 * to get similar blob as in set command. Helps to compare the received
+		 * blob to sent blobs. The comp_id is not useful.
+		 */
+		ctl_data->buffer[BUFFER_TAG_OFFSET] = SOF_CTRL_CMD_BINARY;
```

Comment:
@ujfalusi What do you think of this kernel originated value rewrite?

Reply:
In IPC3 the `header.numid` was used to store the `command` (or type) for the control (VOLUME, SWITCH, BINARY, ENUM), this is not relevant for IPC4, it is largely ignored, but when you get the control data, the kernel sets it to 'something' close enough to numid: `header.numid = scontrol->comp_id`.
I guess we could change the kernel to set this to `SOF_CTRL_CMD_BINARY` for IPC4?

Reply:
Yep, I think it would be good, then the blob get responses for same blobs but from different component instances would be the same.